subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/LauncherServices.java

```diff
@@ -69,19 +73,48 @@ public void registerProjectServices(ServiceRegistration registration) {
     }
 
     static class ToolingGlobalScopeServices {
-        BuildExecuter createBuildExecuter(GradleLauncherFactory gradleLauncherFactory, ServiceRegistry globalServices, ListenerManager listenerManager, FileWatcherFactory fileWatcherFactory, ExecutorFactory executorFactory, StyledTextOutputFactory styledTextOutputFactory, GradleUserHomeScopeServiceRegistry userHomeServiceRegistry) {
-            List<BuildActionRunner> buildActionRunners = globalServices.getAll(BuildActionRunner.class);
+
+        // Can't expose SubscribableBuildActionRunner directly as the call to globalServices.getAll(BuildActionRunner.class)
+        // then fail with a StackOverflowError
+        SubscribableBuildActionRunnerProvider createSubscribableBuildActionRunnerProvider(final ServiceRegistry globalServices) {
```

Comment:
Instead of creating another type, I'd just call this method directly from createBuildExecutor.